sycl/CMakeLists.txt

```diff
@@ -329,6 +329,14 @@ if (LLVM_ENABLE_ASSERTIONS AND NOT SYCL_DISABLE_STL_ASSERTIONS AND NOT WIN32)
   endif()
 endif()
 
+# Need zstd for device image compression.
```

Comment:
Why don't use `LLVM_ENABLE_ZSTD` configured via `configure.py`?

Don't you think we should support all `LLVM_ENABLE_ZSTD` values?

> Used to decide if LLVM tools should support compression/decompression with zstd. Allowed values are OFF, ON (default, enable if zstd is found), and FORCE_ON (error if zstd is not found).

Reply:
> Why don't use LLVM_ENABLE_ZSTD configured via configure.py?

Yup, I could use LLVM_ENABLE_ZSTD. Fixed in https://github.com/intel/llvm/pull/15124/commits/2eb0c25680be2597bb0e5a0d00ec4e1f0f3da5e0

> Don't you think we should support all LLVM_ENABLE_ZSTD values?
> 
> Used to decide if LLVM tools should support compression/decompression with zstd. Allowed values are OFF, ON (default, enable if zstd is found), and FORCE_ON (error if zstd is not found).

My understanding is that during configuration, the user assigns ON/OFF/FORCE_ON value to LLVM_ENABLE_ZSTD, but, after `find_package(zstd)` LLVM_ENABLE_ZSTD gets reassigned with TRUE/FALSE values (here: https://github.com/intel/llvm/blob/sycl/llvm/cmake/config-ix.cmake#L163). That's why, we can directly use `if (LLVM_ENABLE_ZSTD)` in CMakeFiles instead of `if (LLVM_ENABLE_ZSTD STREQUAL "FORCE_ON")` - LLVM also does the same (For example: https://github.com/intel/llvm/blob/sycl/llvm/lib/Support/Compression.cpp#L23, https://github.com/intel/llvm/blob/sycl/llvm/lib/Support/CMakeLists.txt#L28)

Reply:
Right. That's what I meant. Previous implementation implemented only "ON" value. New implementation also supports OFF and FORCE_ON values. 👍
I thought that you will just create a SYCL alias for use in the SYCL project:
```cmake
set(SYCL_ZSTD_AVAILABLE ${LLVM_ENABLE_ZSTD})
```